# HG changeset patch
# User Xerxes Ranby <xerxes@zafena.se>
# Date 1312882468 -7200
# Node ID 82093e560de226787b4c4dca51ad95aaa73a3482
# Parent  7cdc3acdda5b3eab0535c59cd348966a5b825d96
CACAO: Updated to 2011-08-08 revision.

2011-08-09  Xerxes Rånby  <xerxes@zafena.se>

	CACAO
	- CA159: Exception handler blocks / register mixup.
	- Set thread to RUNNABLE during Thread.start.
	- Removed state-setting function call that would be done by the thread
	  itself, creating a nasty race.
	* NEWS: Updated.
	* Makefile.am
	(CACAO_VERSION): Updated CACAO to 2011-08-08 revision.
	(CACAO_SHA256SUM): Updated.

diff -r 7cdc3acdda5b -r 82093e560de2 ChangeLog
--- a/ChangeLog	Fri Sep 30 04:25:29 2011 +0100
+++ b/ChangeLog	Tue Aug 09 11:34:28 2011 +0200
@@ -1,3 +1,15 @@
+2011-08-09  Xerxes Rånby  <xerxes@zafena.se>
+
+	CACAO
+	- CA159: Exception handler blocks / register mixup.
+	- Set thread to RUNNABLE during Thread.start.
+	- Removed state-setting function call that would be done by the thread
+	  itself, creating a nasty race.
+	* NEWS: Updated.
+	* Makefile.am
+	(CACAO_VERSION): Updated CACAO to 2011-08-08 revision.
+	(CACAO_SHA256SUM): Updated.
+
 2010-11-19  Andrew John Hughes  <ahughes@redhat.com>
 
 	* Makefile.am:
diff -r 7cdc3acdda5b -r 82093e560de2 Makefile.am
--- a/Makefile.am	Fri Sep 30 04:25:29 2011 +0100
+++ b/Makefile.am	Tue Aug 09 11:34:28 2011 +0200
@@ -18,8 +18,8 @@
 LANGTOOLS_SHA256SUM = 9ddc00ec50fd2f5e331dc2bc10da4e23b69bf644eb92d50b39a2003c18fb5aa1
 OPENJDK_SHA256SUM = 4043a75c2c4385dd735f8dbbf2369311ce1b951217c9dbe9bba9609e24eb291e
 
-CACAO_VERSION = d6264eb66506
-CACAO_SHA256SUM = 94ea7899e806ccbc33a732b5113a8f969d8b1f4ce7ffd27cf04577054f65f63c
+CACAO_VERSION = 12d5cd5d2dfa
+CACAO_SHA256SUM = f4a93710709f43e203d79c1c8bb73f14ebcca41be75887fb92308b86747312c5
 CACAO_BASE_URL = http://icedtea.classpath.org/download/drops/cacao
 CACAO_URL = $(CACAO_BASE_URL)/$(CACAO_VERSION).tar.gz
 CACAO_SRC_ZIP = cacao-$(CACAO_VERSION).tar.bz2
diff -r 7cdc3acdda5b -r 82093e560de2 NEWS
--- a/NEWS	Fri Sep 30 04:25:29 2011 +0100
+++ b/NEWS	Tue Aug 09 11:34:28 2011 +0200
@@ -107,6 +107,10 @@
   - MIPS: 64-bit and interpreter inlining by default.
   - Trivial implementation of stubs for MIPS.
 * CACAO
+  - CA159: Exception handler blocks / register mixup.
+  - Set thread to RUNNABLE during Thread.start.
+  - Removed state-setting function call that would be done by the thread itself,
+    creating a nasty race.
   - Threadlist & threadobject improvements.
   - Ignore all unknown options, but report them.
   - Fixes build for newer gcc (at least 4.4) on PPC64, breaks older gcc.
